rslearn/data_sources/climate_data_store.py

```diff
@@ -0,0 +1,219 @@
+
+        Returns:
+            A new ERA5LandMonthlyMeans instance
+        """
+        if config.data_source is None:
+            raise ValueError("data_source is required")
+        d = config.data_source.config_dict
+        return ERA5LandMonthlyMeans(
+            config=config,
+            api_key=d["api_key"],
+        )
+
+    def get_items(
+        self, geometries: list[STGeometry], query_config: QueryConfig
+    ) -> list[list[list[Item]]]:
+        """Get a list if items in the data source intersecting the given geometries.
+
+        Args:
+            geometries: the spatiotemporal geometries
+            query_config: the query configuration
+
+        Returns:
+            List of groups of items that should be retrieved for each geometry.
+        """
+        wgs84_geometries = [
+            geometry.to_projection(WGS84_PROJECTION) for geometry in geometries
+        ]
+        groups = []
+        for geometry in wgs84_geometries:
+            cur_items = []
+            # Create an item for each year & month within the time range
+            start_date = geometry.time_range[0].replace(day=1)  # type: ignore
+            end_date = geometry.time_range[1].replace(day=1)  # type: ignore
+            while start_date <= end_date:
+                # Get the first and last day of the current year & month
+                # Use this as the item's time range
+                item_start_date, item_end_date = (
+                    start_date,
+                    start_date + relativedelta(months=1) - relativedelta(days=1),
+                )
+                item_geometry = STGeometry(
+                    geometry.projection, geometry.shp, (item_start_date, item_end_date)
+                )
+                item_bounds = item_geometry.shp.bounds
+                item_name = f"{item_bounds[0]}_{item_bounds[1]}_{item_start_date.year}_{item_start_date.month:02d}"
+                item = Item(item_name, item_geometry)
+                cur_items.append(item)
+                start_date += relativedelta(months=1)
+            cur_groups = match_candidate_items_to_window(
+                geometry, cur_items, query_config
+            )
+            groups.append(cur_groups)
+
+        return groups
+
+    def _convert_nc_to_tif(self, nc_path: UPath, tif_path: UPath) -> None:
+        """Convert a netCDF file to a GeoTIFF file.
+
+        Args:
+            nc_path: the path to the netCDF file
+            tif_path: the path to the output GeoTIFF file
+        """
+        nc = netCDF4.Dataset(nc_path)
+        bands_data = []
+        for band_name in nc.variables:
+            band_data = nc.variables[band_name]
+            # Variable data is stored in a 3D array (1, height, width)
+            if len(band_data.shape) == 3:
+                bands_data.append(band_data[0, :, :])
+        array = np.array(bands_data)  # (num_bands, height, width)
+        # Get metadata for the GeoTIFF
+        lat = nc.variables["latitude"][:]
```

Comment:
Are the latitudes and longitudes always evenly spaced by exactly 0.1 degrees? Otherwise this code may not work correctly.

Reply:
If it is not evenly spaced then you may need to use something like https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.griddata.html#scipy.interpolate.griddata to do bilinear interpolation onto a grid with fixed pixel size first, then compute the transform from the bounds of that interpolated grid.

Reply:
The data on CDS has been regridded to a regular lat-lon grid of 0.1° x 0.1° (reference: https://cds.climate.copernicus.eu/datasets/reanalysis-era5-land-monthly-means?tab=overview), I also downloaded an example nc file, here're how its latitude and longitude look like: 
`<xarray.DataArray 'latitude' (latitude: 61)> Size: 488B
array([46. , 45.9, 45.8, 45.7, 45.6, 45.5, 45.4, 45.3, 45.2, 45.1, 45. , 44.9,
       44.8, 44.7, 44.6, 44.5, 44.4, 44.3, 44.2, 44.1, 44. , 43.9, 43.8, 43.7,
       43.6, 43.5, 43.4, 43.3, 43.2, 43.1, 43. , 42.9, 42.8, 42.7, 42.6, 42.5,
       42.4, 42.3, 42.2, 42.1, 42. , 41.9, 41.8, 41.7, 41.6, 41.5, 41.4, 41.3,
       41.2, 41.1, 41. , 40.9, 40.8, 40.7, 40.6, 40.5, 40.4, 40.3, 40.2, 40.1,
       40. ])`
`<xarray.DataArray 'longitude' (longitude: 21)> Size: 168B
array([-116. , -115.9, -115.8, -115.7, -115.6, -115.5, -115.4, -115.3, -115.2,
       -115.1, -115. , -114.9, -114.8, -114.7, -114.6, -114.5, -114.4, -114.3,
       -114.2, -114.1, -114. ])`
which strictly have the 0.1° interval. 
![Screenshot 2024-12-05 at 6 10 37 PM](https://github.com/user-attachments/assets/44452c23-203b-446e-bb93-783faef019f2)

Reply:
Ah ok great! Not absolutely necessary but minor suggestion to add a check that raises ValueError in case the spacing is not as expected.

Reply:
make sense! I have added the spacing check for both latitude and longitude grids.